apps/percona/functional/snapshot_rebuild/test.yml

```diff
@@ -0,0 +1,189 @@
+          shell: >
+            kubectl exec -it {{ cstor_target.stdout }} -n {{ operator_ns }} --container cstor-istgt 
+            -- cat /usr/local/etc/istgt/istgt.conf | grep -A8 "{{ pv.stdout }}" | grep ConsistencyFactor| cut -d ' ' -f 4
+          args:
+            executable: /bin/bash
+          register: consistency_factor
+
+        - name: Checking the number of healthy pods.
+          shell: >
+            kubectl exec -it {{ cstor_target.stdout }} -n {{ operator_ns }} --container  cstor-istgt 
+            -- istgtcontrol -q replica |json_pp |grep "\"replicaId" |awk -F ':' '{print $2}' | tr -d ',' |wc -l
+          args:
+            executable: /bin/bash
+          register: healthy_pods
+
+        - name: Create an empty list of replica pods.
+          set_fact:
+            cstor_replicas_pod : []
+
+        - name: Obtaining the pool deployments from cvr
+          shell: >
+            kubectl get cvr -n {{ operator_ns }}
+            -l openebs.io/persistent-volume={{ pv.stdout }} --no-headers
+            -o=jsonpath='{range .items[*]}{.metadata.labels.cstorpool\.openebs\.io\/name}{"\n"}{end}'
+          args:
+            executable: /bin/bash
+          register: pool_deployment
+
+        - name: Obtaining the replicasets corresponding to pool deployements.
+          shell: >
+            kubectl get rs --selector=app=cstor-pool -n {{ operator_ns }} --no-headers
+            -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name=="{{item}}")].metadata.name}'
+          register: rs_list
+          with_items:
+            - "{{ pool_deployment.stdout_lines }}"
+
+        - name: Obtaining the pool pods
+          shell: >
+            kubectl get pod --selector=app=cstor-pool -n {{ operator_ns }} --no-headers
+            -o=jsonpath='{.items[?(@.metadata.ownerReferences[0].name=="{{item.stdout}}")].metadata.name}'
+          register: pool_pods
+          with_items:
+            - "{{ rs_list.results }}"
+
+        - name: Build a list of replica pods.
+          set_fact:
+            cstor_replicas_pod : "{{ cstor_replicas_pod }} + [ '{{ item.stdout }}' ]"
+          with_items: "{{ pool_pods.results }}"
+
+        # #Select any cstor replica pod to inject packet loss
+        - set_fact:
+            replica_pod: "{{ cstor_replicas_pod | list | first }}"
+
+        - block:
+
+        ## Inducing the packet drop on the above pool pod.
+            - include_tasks: "/common/utils/inject_packet_loss_tc.yml"
+              vars:
+                status: "induce"
+                target_pod: "{{ replica_pod }}"
+                operator_namespace: "{{ operator_ns }}"
+
+            - name: Check if the targeted replica is disconnected.
+              shell: >
+                kubectl exec -it {{ cstor_target.stdout }} -n {{ operator_ns }} --container  cstor-istgt 
+                -- istgtcontrol -q replica |json_pp |grep "\"replicaId" |awk -F ':' '{print $2}' | tr -d ',' |wc -l
```

Comment:
this also can be changed like above comment to find healthy pods, and it can be verified for previous healthy pod count - 1

Reply:
Also, for the replica, in which packet loss is injected, to get disconnected, IOs should happen. Here, does IOs happen from liveness checks?

Reply:
liveness probe should help..however, that is a deployment litmusbook param - of which we may not be sure while executing 'this' test. Maybe good idea to run  I/O as part of this test itself?